python/mdvtools/stress_tests/test_json_write.py

```
@@ -0,0 +1,264 @@
+    
+    def resource_constrained_test(self, use_atomic=False):
+        """Test writing while system resources are constrained"""
+        test_file = self.test_dir / "resource.json"
+        success_count = 0
+        
+        # Create resource competition with many parallel operations
+        def resource_hog():
+            # Create CPU and I/O pressure
+            temp_files = []
+            for _ in range(10):
+                tf = tempfile.NamedTemporaryFile(dir=self.test_dir, delete=False)
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)